### PR TITLE
fix missing configuration to use the CoreBundle's FormHelper

### PR DESCRIPTION
**Pull request by rande**
I am targetting this branch, because it is the stable branch.

## Changelog

```markdown
### Fixed
- add missing configuration to make the CoreBundle's FormHelper works properly
```

## Subject

Fix missing code to make the FormHelper works properly, this affect the sonata's REST API.

### CODE REVIEW

**Comment by greg0ire** [comment]
Why wasn't it working?

**Comment by rande** [comment]
@greg0ire The Form component (2.8 and upper) uses the fqdn as a key to store the object type. So the current code fix the key duplication.

**Comment by greg0ire** [comment]
Ok. Suggested commit message for squash merging or amending : 

```
Avoid form type name conflict

The Form component (2.8 and upper) uses the FQDN as a key to store the object type.
To avoid a conflict when registering the same type with different names, we create a child class for each name.
```

LGTM

File: Resources/config/api_form.xml [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+        <service id="sonata.news.api.form.type.comment" class="Sonata\NewsBundle\Form\Type\ApiCommentType">
             <tag name="form.type" alias="sonata_news_api_form_comment"/>
             <argument type="service" id="jms_serializer.metadata_factory"/>
             <argument type="service" id="doctrine"/>
             <argument>sonata_news_api_form_comment</argument>
             <argument>%sonata.news.admin.comment.entity%</argument>
             <argument>sonata_api_write</argument>
         </service>
-        <service id="sonata.news.api.form.type.post" class="Sonata\CoreBundle\Form\Type\DoctrineORMSerializationType">
+
```
Comment (greg0ire) [inline Resources/config/api_form.xml]: `xmllint` doesn't like this line :(

**Comment by rande** [comment]
@OskarStark you can also merge this PR ;)

File: Form/Type/ApiCommentType.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Form\Type;
+
+use Sonata\CoreBundle\Form\Type\BaseDoctrineORMSerializationType;
+
+class ApiCommentType extends BaseDoctrineORMSerializationType
```
Comment (core23) [inline Form/Type/ApiCommentType.php]: Please add a short PHPdoc with `@author`

File: Form/Type/ApiPostType.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Form\Type;
+
+use Sonata\CoreBundle\Form\Type\BaseDoctrineORMSerializationType;
+
+class ApiPostType extends BaseDoctrineORMSerializationType
+{
```
Comment (core23) [inline Form/Type/ApiPostType.php]: PHPdoc missing

File: Form/Type/ApiCommentType.php (hunk 2 above)
Comment (rande) [inline Form/Type/ApiCommentType.php]: Arf, should I add my name on 90% of the sonata code base ?

Comment (rande) [inline Form/Type/ApiCommentType.php]:
```php
/**
 * Class ApiCommentType
 *
 * @package Sonata\NewsBundle\Form\Type
 * @author  Thomas Rabaix <thomas.rabaix@gmail.com>
 */
```

Is this comment valid ?

Comment (Soullivaneuh) [inline Form/Type/ApiCommentType.php]: `@package` is not used according to the Symfony convention.

And "Class ApiCommentType" is not really useful. :stuck_out_tongue: 

The `@author` tag should be enough. :wink:

**Comment by rande** [comment]
@core23 I just update the PHPDoc. Thanks

File: Form/Type/ApiCommentType.php (hunk 2 above)
Comment (core23) [inline Form/Type/ApiCommentType.php]: > Arf, should I add my name on 90% of the sonata code base ?

You could, if you want :trollface: But we wanna add the tag to all new classes, just to have general workflow.

**Pull request merged**